app/src/main/java/com/duckduckgo/app/browser/webview/MaliciousSiteBlockerWebViewIntegration.kt

```diff
@@ -45,23 +51,28 @@ interface MaliciousSiteBlockerWebViewIntegration {
     suspend fun shouldIntercept(
         request: WebResourceRequest,
         documentUri: Uri?,
-        confirmationCallback: (isMalicious: Boolean) -> Unit,
-    ): WebResourceResponse?
+        confirmationCallback: (maliciousStatus: MaliciousStatus) -> Unit,
+    ): IsMaliciousViewData
 
     fun shouldOverrideUrlLoading(
         url: Uri,
         isForMainFrame: Boolean,
-        confirmationCallback: (isMalicious: Boolean) -> Unit,
-    ): Boolean
+        confirmationCallback: (maliciousStatus: MaliciousStatus) -> Unit,
+    ): IsMaliciousViewData
 
     fun onPageLoadStarted()
 
-    fun onSiteExempted(url: Uri)
+    fun onSiteExempted(
+        url: Uri,
+        feed: Feed,
+    )
 }
 
+data class ExemptedUrl(val url: Uri, val feed: Feed)
+
 @SingleInstanceIn(AppScope::class)
 class ExemptedUrlsHolder @Inject constructor() {
-    val exemptedMaliciousUrls = mutableSetOf<String>()
+    val exemptedMaliciousUrls = mutableSetOf<ExemptedUrl>()
```

Comment:
When checking how we use this, I don't see we ever check feed, only url. Do we have plans to differentiate exemptedUrls per feed?

Reply:
We don't check it here, but we need to be able to return both the URL and the feed, so that we can then setup the privacy dashboard for exempted sites, as the message shown is different depending on the feed that flagged the site